src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/CLinker.java

```diff
@@ -54,132 +96,430 @@ private CSupport() {}
      * @throws IllegalAccessError if the runtime property {@code foreign.restricted} is not set to either
      * {@code permit}, {@code warn} or {@code debug} (the default value is set to {@code deny}).
      */
-    public static ForeignLinker getSystemLinker() {
-        Utils.checkRestrictedAccess("CSupport.getSystemLinker");
+    static CLinker getSystemLinker() {
```

Comment:
Should be `getSystemLinker` here? Also, the string passed to the restricted check doesn't match the name of the method

Reply:
Not sure what you mean here? The method is named getSystemLinker?

I've changed the message text.

Reply:
Discussed offline: getSystemLinker implies too much that a particular system has One True Linker™. Changing this name to getInstance().